### PR TITLE
Add dsl2 module for deepvariant

### PR DESCRIPTION
**Pull request by abhi18av**
Hi team,

I'm initiating the draft PR for the integration of `deepvariant` tool, we had a discussion a while back in the nf-core's #deepvariant channel regarding this but for I couldn't find the time to take it forward then. Now, I find myself with some time to finally see this through.

I'll update the checklist once I have finalized the code and it's ready for a wider/final review. In the meantime, I've marked this PR as a draft to have a placeholder for early discussions

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
    - [x] If you've added a new tool - add to the software_versions process and a regex to `scrape_software_versions.py`
    - [x] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)
    - [ ] If necessary, also make a PR on the nf-core/sarek _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint .`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: subworkflows/local/germline_variant_calling.nf [hunk 1]
```diff
@@ -94,6 +96,17 @@ workflow GERMLINE_VARIANT_CALLING {
         haplotypecaller_vcf = CONCAT_HAPLOTYPECALLER.out.vcf
     }
 
+    if ('deepvariant' in params.tools.toLowerCase()) {
```
Comment (abhi18av) [inline subworkflows/local/germline_variant_calling.nf:99]: @maxulysse , please correct me if I'm wrong in assuming that this is how I could integrate the WIP module?

File: subworkflows/local/germline_variant_calling.nf [hunk 2]
```diff
@@ -94,6 +96,17 @@ workflow GERMLINE_VARIANT_CALLING {
...
+
+        DEEPVARIANT(
+            bam,
```
Comment (abhi18av) [inline subworkflows/local/germline_variant_calling.nf:102]: Based on the code for `bam` https://github.com/nf-core/sarek/blob/88ac395f1261d66c2bca6032f48d02ce69d44231/workflows/sarek.nf#L462   ... I am assuming that this is of the following shape 

```nextflow
    tuple val(meta), path(bam), path(bai)
```

Is this assumption correct?

File: modules/local/deepvariant/functions.nf [hunk 3]
```diff
@@ -0,0 +1,70 @@
+/*
```
Comment (abhi18av) [inline modules/local/deepvariant/functions.nf:1]: Haven't made any changes in the `functions.nf` file - is any customization required here?

**Review by abhi18av — COMMENTED** [review]
I've added some comments to confirm certain design decisions and request feedback.

I'm a bit confused about how to use the `options.args` to pass arbitrary args to the workflow, could you please point me to the right direction?


Also, is there a way to skip triggering the CI for these interim changes?

File: subworkflows/local/germline_variant_calling.nf (hunk 1 above)
Comment (maxulysse) [reply subworkflows/local/germline_variant_calling.nf:99]: That's perfect

File: modules/local/deepvariant/functions.nf (hunk 3 above)
Comment (maxulysse) [reply modules/local/deepvariant/functions.nf:1]: Nothing needs to be done here

File: subworkflows/local/germline_variant_calling.nf (hunk 2 above)
Comment (maxulysse) [reply subworkflows/local/germline_variant_calling.nf:102]: Totally

**Comment by maxulysse** [comment]
> I've added some comments to confirm certain design decisions and request feedback.
> 
> I'm a bit confused about how to use the `options.args` to pass arbitrary args to the workflow, could you please point me to the right direction?

You need to add the args (args2 or even args3) to https://github.com/nf-core/sarek/blob/dsl2/conf/modules.config

> Also, is there a way to skip triggering the CI for these interim changes?

Add skip ci to your commit:

https://github.blog/changelog/2021-02-08-github-actions-skip-pull-request-and-push-workflows-with-skip-ci/

**Comment by abhi18av** [comment]
Hi @maxulysse , I've now opened up the PR for review

- implemented the basic module 
- intergrated the settings into `modules.config`
- integrated the module into `germline_variant_caller.nf` subworkflow

Please let me know what further changes are necessary before this PR could be mergeable.

**Review by maxulysse — APPROVED** [review]

File: subworkflows/local/germline_variant_calling.nf [hunk 4]
```diff
@@ -94,6 +96,17 @@ workflow GERMLINE_VARIANT_CALLING {
...
+            fasta,
+            fai)
+
+        haplotypecaller_vcf = DEEPVARIANT.out.vcf
+        haplotypecaller_gvcf = DEEPVARIANT.out.gvcf
```
Comment (maxulysse) [inline subworkflows/local/germline_variant_calling.nf:107]:
```suggestion
        deepvariant_vcf = DEEPVARIANT.out.vcf
        deepvariant_gvcf = DEEPVARIANT.out.gvcf
```

Comment (maxulysse) [reply subworkflows/local/germline_variant_calling.nf]: doesn't make much sense to have output from deepvariant to go in haplotypecaller channel

**Comment by abhi18av** [comment]
Thanks @maxulysse !  😍

**Pull request merged**